lib/puppet/pops/types/type_mismatch_describer.rb

```diff
@@ -581,6 +581,11 @@ def validate_default_parameter(subject, param_name, param_type, value, tense = :
       end
     end
 
+    def get_deferred_function_return_type(value)
+     func =  Puppet::Pops::Loaders.find_loader().load_typed(Puppet::Pops::Loader::TypedName.new(:function, value.name)).value
+     func.class.dispatcher.dispatchers[0].type.return_type
```

Comment:
This is also super crude in that it only introspects the first `dispatcher` in array; I need to go back and get a better understanding on how to iterate through the different arrities. Instead of just know the function name, we might need to know the type of each argument as well. I am assuming puppet can specify different return_types based on different arrities...